gems/smithy/lib/smithy/anvil/client/templates/plugins/endpoint.erb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# This is generated code!
+
+module <%= namespace %>
+  module Plugins
+    class Endpoint < Smithy::Client::Plugin
+      option(
+        :endpoint_provider,
+        doc_type: '<%= namespace %>::EndpointProvider',
+        docstring: <<~DOCS) do |_cfg|
+The endpoint provider used to resolve endpoints. Any object that responds to
+`#resolve_endpoint(parameters)`.
+        DOCS
+        EndpointProvider.new
```

Comment:
This could just be passed as `default:` instead of as the block?

Reply:
I think that will create one object and not a new one each time?

Reply:
What do you mean? The block or default in config options is run once in client initialize

Reply:
The default operation (if its not a block) is only ever run once (ie, at the time the file is required), which means multiple instances of a class will share the same object here, so it must be a block.

Reply:
Ah ok